### ISSUE TITLE
the run method mutates the provided fact

### ISSUE THREAD

**Issue by antoniopresto** [CacheControl/json-rules-engine#187]
https://github.com/CacheControl/json-rules-engine/blob/master/src/engine.js#L213

I'll send a PR.

**Comment by CacheControl** [comment]
@antoniopresto Is this causing a problem, or is this a general "remove mutation to improve code base health" kind of issue?

**Comment by antoniopresto** [comment]
this is  causing a problem, adding one unknown field to my form state.

**Comment by CacheControl** [comment]
@antoniopresto were you able to figure this out? At first I thought your comment meant that there was a bug where the success-events facts was returning an  `undefined` value, but having reviewed the code I don't see how that's possible, since it's initialized as an empty array. Perhaps you were adding all the runtime facts to your form state, and seeing the `success-events` fact appear was unexpected?

**Comment by antoniopresto** [comment]
Sorry @CacheControl, I was very busy when I wrote the description and it looks really bad.

The problem is that the run method is modifying the original received as `runtimeFacts`.

ex:
 ```js
const myFormState = { name: 'Antonio' };

await engine.run(myFormState)

console.log(Object.keys(myFormState)); // ['name', 'success-events'] <-- 'success-events' was added 
```

**Comment by dumbape** [comment]
Any progress on this?

**Comment by calpa** [comment]
This cause `Uncaught TypeError: Cannot add property success-events, object is not extensible`, which is so strange...

**Comment by CacheControl** [comment]
This bug is fixed in **v6.0.0**. See [changelog](https://github.com/CacheControl/json-rules-engine/blob/master/CHANGELOG.md#60--2020-12-22)
